zrml/market-commons/src/migrations.rs

```diff
@@ -14,30 +14,28 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
-
-use crate::{Config, Pallet};
-use alloc::{string::ToString, vec::Vec};
+use crate::{Config, MarketIdOf, MomentOf};
 use frame_support::{
     log,
     migration::{put_storage_value, storage_iter},
-    pallet_prelude::{PhantomData, StorageVersion},
+    pallet_prelude::PhantomData,
     traits::{Get, OnRuntimeUpgrade},
-    RuntimeDebug,
 };
-use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
-#[cfg(feature = "try-runtime")]
-use scale_info::prelude::format;
-use scale_info::TypeInfo;
+extern crate alloc;
+use alloc::vec::Vec;
+use parity_scale_codec::{Decode, Encode};
 use zeitgeist_primitives::types::{
-    Deadlines, Market, MarketCreation, MarketDisputeMechanism, MarketPeriod, MarketStatus,
+    Asset, Deadlines, Market, MarketCreation, MarketDisputeMechanism, MarketPeriod, MarketStatus,
     MarketType, OutcomeReport, Report, ScoringRule,
 };
+
+const MARKET_COMMONS_REQUIRED_STORAGE_VERSION: u16 = 4;
+const MARKET_COMMONS_NEXT_STORAGE_VERSION: u16 = 5;
+
 const MARKET_COMMONS: &[u8] = b"MarketCommons";
 const MARKETS: &[u8] = b"Markets";
-const MARKET_COMMONS_REQUIRED_STORAGE_VERSION: u16 = 2_u16;
-const MARKET_COMMONS_NEXT_STORAGE_VERSION: u16 = 3_u16;
 
-#[derive(Clone, Decode, Encode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+#[derive(Clone, Decode, Encode)]
```

Comment:
Currently there are these derives for a market https://github.com/zeitgeistpm/zeitgeist/blob/f8f5c79c7c49c9011ab00ffa670d40dab48d44b7/primitives/src/market.rs#L30

Reply:
but do we really need other traits derived ? I have taken reference from https://github.com/zeitgeistpm/zeitgeist/pull/732/files#diff-7eef70e0e24e29abbdd8ae4c5949b3d7633f1fa7b844e9b0e7ff3227cc344865R50 and it works fine in the test migration.

Reply:
I must admit that I don't know an answer to this question. So if you have the proof that it works, we should ignore this.